json-core/src/main/java/io/micronaut/json/codec/MapperMediaTypeCodec.java

```diff
@@ -264,8 +265,15 @@ public <T, B> ByteBuffer<B> encode(Argument<T> type, T object, ByteBufferFactory
             return allocator.copiedBuffer((byte[]) object);
         }
         ByteBuffer<B> buffer = allocator.buffer();
-        OutputStream outputStream = buffer.toOutputStream();
-        encode(type, object, outputStream);
-        return buffer;
+        try {
+            OutputStream outputStream = buffer.toOutputStream();
+            encode(type, object, outputStream);
+            return buffer;
+        } catch (Throwable t) {
```

Comment:
Sonar recommends catching `Exception` instead of `Throwable`. Should we do that?

Reply:
no, this is supposed to work kind of like a finally block, it should always release